charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -246,21 +246,8 @@ spec:
   ruleSelector: {}
 {{- end }}
 {{- end }}
-{{- if .Values.prometheus.prometheusSpec.scrapeConfigSelector }}
-  scrapeConfigSelector:
-{{ tpl (toYaml .Values.prometheus.prometheusSpec.scrapeConfigSelector | indent 4) . }}
-{{ else if .Values.prometheus.prometheusSpec.scrapeConfigSelectorNilUsesHelmValues  }}
-  scrapeConfigSelector:
-    matchLabels:
-      release: {{ $.Release.Name | quote }}
-{{ else }}
-  scrapeConfigSelector: {}
-{{- end }}
-{{- if .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector }}
-  scrapeConfigNamespaceSelector:
-{{ tpl (toYaml .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector | indent 4) . }}
-{{ else }}
-  scrapeConfigNamespaceSelector: {}
+{{- if not (kindIs "invalid" .Values.prometheus.prometheusSpec.scrapeConfigSelector) }}
+  scrapeConfigSelector: {{ tpl (toYaml .Values.prometheus.prometheusSpec.scrapeConfigSelector) . | nindent 4 }}
```

Comment:
scrapeConfigSelector and scrapeConfigNamespaceSelector are distinct properties.

Reply:
Good catch! Brought it back in 0b8eb3c1 and will convert it with the other properties if enough maintainers agree.